### PR TITLE
QA bulk update

### PR DESCRIPTION
**Pull request by aapomm**
### Summary

Provide a general description of the code changes in your pull
request... were there any bugs you had fixed? If so, mention them. If
these bugs have open GitHub issues, be sure to tag them here as well,
to keep the conversation linked together.

### Other Information

If there's anything else that's important and relevant to your pull
request, mention that information here. This could include
benchmarks, or other information.

Thanks for contributing to Dradis!

### Copyright assignment

Collaboration is difficult with commercial closed source but we want
to keep as much of the OSS ethos as possible available to users
who want to fix it themselves.

In order to unambiguously own and sell Dradis Framework commercial
products, we must have the copyright associated with the entire
codebase. Any code you create which is merged must be owned by us.
That's not us trying to be a jerks, that's just the way it works.

Please review the [CONTRIBUTING.md](https://github.com/dradis/dradis-ce/blob/master/CONTRIBUTING.md)
file for the details.

You can delete this section, but the following sentence needs to
remain in the PR's description:

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### Check List

- [ ] Added a CHANGELOG entry

### CODE REVIEW

File: app/assets/javascripts/shared/datatable/state.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        that.toggleLoadingState(selectedRows, false);
+        that.toggleStateBtn(false);
+
+        selectedRows.deselect().remove().draw();
+
+        $('[data-behavior="qa-alert"]').remove();
+        $('.page-title').after(`
+          <div class="alert alert-success" data-behavior="qa-alert">
+            <a class="close" data-dismiss="alert" href="javascript:void(0)">x</a>
+            Successfully set the issues as ${newState}!
```
Comment (MattBudz) [inline app/assets/javascripts/shared/datatable/state.js:42]:
```suggestion
            <p>Successfully set the issues as ${newState}!</p>
```

We should have text directly in a div

Comment (aapomm) [reply app/assets/javascripts/shared/datatable/state.js:42]: Looks like this when I add the paragraph tag:

Screenshot 2023-03-10 at 6 42 39 PM

Note that the text is inside the `<div class="alert">` and this is the same structure we use for `flash_messages` and alerts from the datatabel (delete.js).

File: app/controllers/qa/issues_controller.rb [hunk 2]
```diff
@@ -5,4 +5,12 @@ def index
     @issues = current_project.issues.ready_for_review
     @all_columns = @default_columns = ['Title']
   end
+
+  def multiple_update
+    @issues = current_project.issues.ready_for_review.where(id: params[:ids])
```
Comment (etdsoft) [inline app/controllers/qa/issues_controller.rb:10]: Isn't this missing some serious validation?

Comment (etdsoft) [inline app/controllers/qa/issues_controller.rb:10]: Isn't this missing some serious validation?

File: app/controllers/qa/issues_controller.rb [hunk 3]
```diff
@@ -5,4 +5,12 @@ def index
...
+
+    @issues.update_all(state: params[:state].parameterize(separator: '_'))
```
Comment (etdsoft) [inline app/controllers/qa/issues_controller.rb:12]: and here as well.

The use of `params` is a smell /cc @MattBudz

File: config/routes.rb [hunk 4]
```diff
@@ -96,7 +96,9 @@
     resources :tags, except: [:show]
 
     namespace :qa do
-      resources :issues, only: [:index, :show, :edit]
+      resources :issues, only: [:index, :show, :edit] do
+        collection { put :multiple_update }
```
Comment (etdsoft) [inline config/routes.rb:100]: This is a routing concern in Pro, and should be here as well for parity.

File: app/assets/javascripts/shared/datatable/main.js [hunk 5]
```diff
@@ -94,6 +94,20 @@ class DradisDatatable {
             text: '<i class="fa fa-tags fa-fw"></i>Tag<i class="fa fa-caret-down fa-fw"></i>',
             buttons: this.setupTagButtons()
           },
+          {
```
Comment (etdsoft) [inline app/assets/javascripts/shared/datatable/main.js:97]: This is polluting all data-tables for something we'll use exactly once.

You can extract this to the  app/assets/javascripts/shared/datatable/state.js and only add it if the DT needs it. See JIRA for an example.

**Review by etdsoft — CHANGES_REQUESTED** [review]

**Pull request merged**